pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt

```diff
@@ -285,7 +303,8 @@ class KotlinCodeGenerator(
 
         // avoid generating multiple methods with same no. of parameters
         if (currParameters.size < prevParameterCount) {
-          val isOverride = currClass !== pClass || superclass != null && properties.isEmpty()
+          val isOverride =
+            currClass !== pClass || nearestNonAbstractAncestorDeclaresCopyMethodWithSameArity()
```

Comment:
[nit] Either make `nearestNonAbstractAncestorDeclaresCopyMethodWithSameArity` a lazy value, or cache this so we don't call it on every iteration

Reply:
It's only called once due to `currClass !== pClass ||`.

Reply:
Ah, yeah, that was a little confusing, but you're right